src/utils/useEscalationColor.ts

```
@@ -0,0 +1,20 @@
+import { useMemo } from 'react';
+import { regionThresholds } from '~/components/choropleth/region-thresholds';
+import { ChoroplethThresholds } from '~/components/choropleth/shared';
+
+export function useEscalationColor(level: number | undefined) {
+  return useMemo(() => {
+    if (level === undefined) {
```

Might be worth doing an assert instead to fail hard. I think you might want to break the app if the data is not compatible with the escalation levels you can handle in the code.

---

I've added another assert as well, in case a color could not be found for the given level